lib/transition-checker.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var List = require('immutable').List;
+var Octokat = require('octokat');
+
+/**
+ * @exports lib/transition-checker
+ */
+
+var TransitionChecker = {};
+
+/**
+ * @returns {Promise.<List.<String>>}
+ */
+
+TransitionChecker.check = function (profile, latestVersion, previousVersion, isEditorial, isUpdate) {
+  return new Promise(function (resolve, reject) {
+    var errors = new List();
+
+    if (profile === 'WD' || profile === 'WG-NOTE' || isUpdate) {
+      resolve({ errors: errors, requiresCfE: false });
+    }
+    else if (profile === 'CR') {
+      if (previousVersion.includes('/WD-') || (previousVersion.includes('/CR-') && !isEditorial)) {
+        var octo = new Octokat({
+          token: global.GH_TOKEN
+        });
+        var repo = octo.repos('w3c', 'transitions');
+
+        var shortname = latestVersion.match(new RegExp(/.*\/([^/]+)\/$/))[1];
```

Comment:
About this regex: are we sure all *latest version* URLs end with a slash?

Reply:
Yes, that's a requirement from pubrules these days.